metals/src/main/scala/scala/meta/internal/metals/logging/MetalsLogger.scala

```diff
@@ -115,34 +117,48 @@ object MetalsLogger {
       folders: List[AbsolutePath],
       redirectSystemStreams: Boolean,
   ): Unit = {
-    val newLogFiles = folders.map(_.resolve(workspaceLogPath))
-    newLogFiles.foreach(truncateLogFile)
+    val newLogFiles = folders.map(backUpOldLogFileIfTooBig)
     scribe.info(s"logging to files ${newLogFiles.mkString(",")}")
     if (redirectSystemStreams) {
       redirectSystemOut(newLogFiles)
     }
   }
 
-  private def truncateLogFile(path: AbsolutePath) = {
-    val MaxLines = 10000
-    if (Files.exists(path.toNIO)) {
+  private def backupLogsDir(workspaceFolder: AbsolutePath) =
+    workspaceFolder.resolve(".metals").resolve(".backup_logs")
+
+  private def backUpOldLogFileIfTooBig(
+      workspaceFolder: AbsolutePath
+  ): AbsolutePath = {
+    val logFilePath = workspaceFolder.resolve(workspaceLogPath)
+    val MAX_SIZE = 3 << 20
+    if (logFilePath.isFile && Files.size(logFilePath.toNIO) > MAX_SIZE) {
+      val backedUpLogFile = backupLogsDir(workspaceFolder).resolve(
+        s"log_${System.currentTimeMillis()}"
+      )
+      backedUpLogFile.parent.createDirectories()
```

Comment:
Would it be a good idea to move this call inside the `try` block below, to handle any IO exceptions? :slightly_smiling_face: